src/pages/SingleArticle.js

```diff
@@ -0,0 +1,67 @@
+import { useParams, Link, useSearchParams } from "react-router-dom";
+import styles from "./singleArticle.module.scss";
+import { NewsContext } from "contexts/news.context";
+import { useContext, useEffect, useState } from "react";
+import apiFetcher from "../helpers/apiFetcher";
+import BodySpinner from "components/BodySpinner";
+
+export default function SingleArticle() {
+  const { news, setNews } = useContext(NewsContext);
+  const { number, id } = useParams();
+  const [searchParams] = useSearchParams();
+  const [isLoading, setisLoading] = useState();
+
+  useEffect(() => {
+    if (news.articles || !searchParams) return;
```

Comment:
You always have searchParams object, so be more precise like searchParams.has("....")